auth.c

```diff
@@ -53,13 +68,95 @@ void janus_auth_deinit(void) {
 	if(allowed_plugins != NULL)
 		g_hash_table_destroy(allowed_plugins);
 	allowed_plugins = NULL;
+	if(auth_secret != NULL)
+		g_free(auth_secret);
+	auth_secret = NULL;
 	janus_mutex_unlock(&mutex);
 }
 
+gboolean janus_auth_check_signature(const char *token, const char *realm) {
+	if (!auth_enabled || auth_secret == NULL)
+		return FALSE;
+	gchar *token_signature = strchr(token, ':') + 1;
+	gchar *token_metadata = g_strndup(token, token_signature - token - 1);
+	gchar **list = g_strsplit(token_metadata, ",", 3);
+	/* need at least an expiry timestamp and realm */
+	if(!list[0] || !list[1])
+		goto fail;
+	/* verify timestamp */
+	int expiry_time = atoi(list[0]);
+	int real_time = janus_get_real_time() / 1000000;
+	if(real_time > expiry_time)
+		goto fail;
+	/* verify realm */
+	if(strcmp(list[1], realm))
+		goto fail;
+	/* verify HMAC-SHA1 */
+	unsigned char signature[EVP_MAX_MD_SIZE];
+	unsigned int len;
+	HMAC(EVP_sha1(), auth_secret, strlen(auth_secret), token_metadata, strlen(token_metadata), signature, &len);
```

Comment:
My gcc here complains about a:

```
auth.c:97:53: warning: pointer targets in passing argument 4 of ‘HMAC’ differ in signedness [-Wpointer-sign]
  HMAC(EVP_sha1(), auth_secret, strlen(auth_secret), token_metadata, strlen(token_metadata), signature, &len);
                                                     ^~~~~~~~~~~~~~
In file included from auth.c:24:0:
/usr/include/openssl/hmac.h:37:16: note: expected ‘const unsigned char *’ but argument is of type ‘gchar * {aka char *}’
 unsigned char *HMAC(const EVP_MD *evp_md, const void *key, int key_len,
                ^~~~
```

It should be enough to cast `(const char *)token_metadata`. Same for the HMAC(..) call at line 142.